pkg/steps/source.go

```diff
@@ -293,6 +297,11 @@ func buildFromSource(jobSpec *api.JobSpec, fromTag, toTag api.PipelineImageStrea
 			},
 		},
 	}
+	if _, err := secretClient.Secrets(jobSpec.Namespace).Get(PullSecretName, meta.GetOptions{}); err == nil {
+		build.Spec.Strategy.DockerStrategy.PullSecret = getSourceSecretFromName(PullSecretName)
+	} else if errors.IsNotFound(err) {
+		log.Printf("Cannot find pull secret '%s', ignoring set it up in the build: %v\n", PullSecretName, err)
```

Comment:
Why is this just a log? Won't it make the build fail?

Reply:
OK ... here is the logic:

Only when `--image-import-pull-secret` is provided to `ci-operator`, we have the secret created. For now we have it provided only for migrated jobs to build01.

So before we migrate all jobs, we will keep the backward-compatibility?

Reply:
So shouldn't we be able to tell when someone does or does not pass that flag, and only do this if they pass it?

Reply:
My first version was like that ... passing the option.pullsecretpath all the way to the changed function.
But i think the current one is better. If it passes the arg, we will create. Then we can get it by client.

Reply:
Updated.